package/MDAnalysis/analysis/base.py

```diff
@@ -94,28 +97,25 @@ def _validate_key(self, key):
     def __init__(self, **kwargs):
         if "data" in kwargs.keys():
             raise AttributeError(f"'data' is a protected dictionary attribute")
-
-        self._dict_frozen = False
-        for key in kwargs:
-            self._validate_key(key)
-        super().__init__(**kwargs)
-        self._dict_frozen = True
+        self.__dict__["data"] = {}
+        self.update(kwargs)
```

Comment:
[`UserDict`](https://github.com/python/cpython/blob/7424105a02bbc209e032752636bae30ccb556402/Lib/collections/__init__.py#L1040-L1048) allow to initlize the instance with passing a dictionary. This does not work anymore. I suggest you copy the four lines 
```python
if dict is not None:
    self.update(dict)
if kwargs:
    self.update(kwargs)
```

from cpython, adjust the arguments, add a test and everything is fine.

Reply:
Are you happier with [my somewhat lazier solution? :) ](https://github.com/MDAnalysis/mdanalysis/pull/3281/commits/619d9374851c4b3c8e45f3a95f0181b080288285)

Reply:
Totally!